{
    "assignee": null,
    "assignees": [],
    "author_association": "NONE",
    "body": "The following is legitimate CPython code:\n\n```\nd = {}\nworks = { k : [n for n in names] for k, names in d.items() }\nfails = { k : set(n for n in names) for k, names in d.items() }\n```\n\nThe \"fails\" line causes the following error:\ncython_defect.py:3:34: local variable 'names' referenced before assignment\n\nThis may have the same root cause as #1351 (based purely on the description).\n",
    "closed_at": "2017-02-07T06:36:00Z",
    "comment_data": [
        {
            "author_association": "CONTRIBUTOR",
            "body": "Another failure that looks related is below (cython v0.25.2). The entire code is in the error message:\r\n\r\n```\r\nError compiling Cython file:\r\n------------------------------------------------------------\r\n...\r\nclass C(object):\r\n  a = {}\r\n  b = list(k for k in a)\r\n                      ^\r\n------------------------------------------------------------\r\n\r\n/tmp/x.py:3:23: undeclared name not builtin: a\r\n```",
            "created_at": "2017-01-24T20:50:54Z",
            "html_url": "https://github.com/cython/cython/issues/1479#issuecomment-274934395",
            "id": 274934395,
            "issue_url": "https://api.github.com/repos/cython/cython/issues/1479",
            "updated_at": "2017-01-24T20:50:54Z",
            "url": "https://api.github.com/repos/cython/cython/issues/comments/274934395",
            "user": {
                "avatar_url": "https://avatars2.githubusercontent.com/u/7066873?v=4",
                "events_url": "https://api.github.com/users/shalabhc/events{/privacy}",
                "followers_url": "https://api.github.com/users/shalabhc/followers",
                "following_url": "https://api.github.com/users/shalabhc/following{/other_user}",
                "gists_url": "https://api.github.com/users/shalabhc/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/shalabhc",
                "id": 7066873,
                "login": "shalabhc",
                "organizations_url": "https://api.github.com/users/shalabhc/orgs",
                "received_events_url": "https://api.github.com/users/shalabhc/received_events",
                "repos_url": "https://api.github.com/users/shalabhc/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/shalabhc/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/shalabhc/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/shalabhc"
            }
        },
        {
            "author_association": "OWNER",
            "body": "Duplicate of #1351.",
            "created_at": "2017-02-07T06:36:00Z",
            "html_url": "https://github.com/cython/cython/issues/1479#issuecomment-277914137",
            "id": 277914137,
            "issue_url": "https://api.github.com/repos/cython/cython/issues/1479",
            "updated_at": "2017-02-07T06:36:00Z",
            "url": "https://api.github.com/repos/cython/cython/issues/comments/277914137",
            "user": {
                "avatar_url": "https://avatars0.githubusercontent.com/u/491659?v=4",
                "events_url": "https://api.github.com/users/scoder/events{/privacy}",
                "followers_url": "https://api.github.com/users/scoder/followers",
                "following_url": "https://api.github.com/users/scoder/following{/other_user}",
                "gists_url": "https://api.github.com/users/scoder/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/scoder",
                "id": 491659,
                "login": "scoder",
                "organizations_url": "https://api.github.com/users/scoder/orgs",
                "received_events_url": "https://api.github.com/users/scoder/received_events",
                "repos_url": "https://api.github.com/users/scoder/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/scoder/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/scoder/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/scoder"
            }
        },
        {
            "author_association": "OWNER",
            "body": "Or rather #1159.",
            "created_at": "2017-08-18T15:02:29Z",
            "html_url": "https://github.com/cython/cython/issues/1479#issuecomment-323377979",
            "id": 323377979,
            "issue_url": "https://api.github.com/repos/cython/cython/issues/1479",
            "updated_at": "2017-08-18T15:02:29Z",
            "url": "https://api.github.com/repos/cython/cython/issues/comments/323377979",
            "user": {
                "avatar_url": "https://avatars0.githubusercontent.com/u/491659?v=4",
                "events_url": "https://api.github.com/users/scoder/events{/privacy}",
                "followers_url": "https://api.github.com/users/scoder/followers",
                "following_url": "https://api.github.com/users/scoder/following{/other_user}",
                "gists_url": "https://api.github.com/users/scoder/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/scoder",
                "id": 491659,
                "login": "scoder",
                "organizations_url": "https://api.github.com/users/scoder/orgs",
                "received_events_url": "https://api.github.com/users/scoder/received_events",
                "repos_url": "https://api.github.com/users/scoder/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/scoder/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/scoder/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/scoder"
            }
        },
        {
            "author_association": "CONTRIBUTOR",
            "body": "This still seems to be a bug in Cython 0.26 but only with `language_level=3`:\r\n\r\n```\r\n$ cat itertest.pyx\r\nxs = [(1, 2), (3, 4), (5, 6)]\r\nprint([sum(a for a in x) for x in xs])\r\n$ cythonize itertest.pyx                       \r\nCompiling itertest.pyx because it changed.\r\n[1/1] Cythonizing itertest.pyx\r\n$ rm -f itertest.c\r\n$ cythonize -3 itertest.pyx\r\nCompiling itertest.pyx because it changed.\r\n[1/1] Cythonizing itertest.pyx\r\n\r\nError compiling Cython file:\r\n------------------------------------------------------------\r\n...\r\nxs = [(1, 2), (3, 4), (5, 6)]\r\nprint([sum(a for a in x) for x in xs])\r\n                      ^\r\n------------------------------------------------------------\r\n\r\nitertest.pyx:2:23: local variable 'x' referenced before assignment\r\nTraceback (most recent call last):\r\n...\r\nCython.Compiler.Errors.CompileError: itertest.pyx\r\n```\r\n\r\nThis code of course works in pure Python 3.",
            "created_at": "2017-09-07T12:50:29Z",
            "html_url": "https://github.com/cython/cython/issues/1479#issuecomment-327789350",
            "id": 327789350,
            "issue_url": "https://api.github.com/repos/cython/cython/issues/1479",
            "updated_at": "2017-09-07T12:50:29Z",
            "url": "https://api.github.com/repos/cython/cython/issues/comments/327789350",
            "user": {
                "avatar_url": "https://avatars1.githubusercontent.com/u/676149?v=4",
                "events_url": "https://api.github.com/users/embray/events{/privacy}",
                "followers_url": "https://api.github.com/users/embray/followers",
                "following_url": "https://api.github.com/users/embray/following{/other_user}",
                "gists_url": "https://api.github.com/users/embray/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/embray",
                "id": 676149,
                "login": "embray",
                "organizations_url": "https://api.github.com/users/embray/orgs",
                "received_events_url": "https://api.github.com/users/embray/received_events",
                "repos_url": "https://api.github.com/users/embray/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/embray/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/embray/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/embray"
            }
        },
        {
            "author_association": "OWNER",
            "body": "I didn't close the ticket because it's fixed but because it's a duplicate. Fixes welcome for #1159.",
            "created_at": "2017-09-07T13:26:19Z",
            "html_url": "https://github.com/cython/cython/issues/1479#issuecomment-327798518",
            "id": 327798518,
            "issue_url": "https://api.github.com/repos/cython/cython/issues/1479",
            "updated_at": "2017-09-07T13:26:19Z",
            "url": "https://api.github.com/repos/cython/cython/issues/comments/327798518",
            "user": {
                "avatar_url": "https://avatars0.githubusercontent.com/u/491659?v=4",
                "events_url": "https://api.github.com/users/scoder/events{/privacy}",
                "followers_url": "https://api.github.com/users/scoder/followers",
                "following_url": "https://api.github.com/users/scoder/following{/other_user}",
                "gists_url": "https://api.github.com/users/scoder/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/scoder",
                "id": 491659,
                "login": "scoder",
                "organizations_url": "https://api.github.com/users/scoder/orgs",
                "received_events_url": "https://api.github.com/users/scoder/received_events",
                "repos_url": "https://api.github.com/users/scoder/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/scoder/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/scoder/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/scoder"
            }
        },
        {
            "author_association": "CONTRIBUTOR",
            "body": "I was a little confused about the status of #1159, since it mentions something about a commit where this was fixed, and indeed it doesn't seem to be a problem for me except with `language_level=3` which that ticket didn't mention...",
            "created_at": "2017-09-07T14:43:18Z",
            "html_url": "https://github.com/cython/cython/issues/1479#issuecomment-327821280",
            "id": 327821280,
            "issue_url": "https://api.github.com/repos/cython/cython/issues/1479",
            "updated_at": "2017-09-07T14:43:18Z",
            "url": "https://api.github.com/repos/cython/cython/issues/comments/327821280",
            "user": {
                "avatar_url": "https://avatars1.githubusercontent.com/u/676149?v=4",
                "events_url": "https://api.github.com/users/embray/events{/privacy}",
                "followers_url": "https://api.github.com/users/embray/followers",
                "following_url": "https://api.github.com/users/embray/following{/other_user}",
                "gists_url": "https://api.github.com/users/embray/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/embray",
                "id": 676149,
                "login": "embray",
                "organizations_url": "https://api.github.com/users/embray/orgs",
                "received_events_url": "https://api.github.com/users/embray/received_events",
                "repos_url": "https://api.github.com/users/embray/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/embray/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/embray/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/embray"
            }
        },
        {
            "author_association": "OWNER",
            "body": "Ah, yes, it was a bit messed up by the initial github import. I've cleaned it up a little.",
            "created_at": "2017-09-07T15:00:32Z",
            "html_url": "https://github.com/cython/cython/issues/1479#issuecomment-327826728",
            "id": 327826728,
            "issue_url": "https://api.github.com/repos/cython/cython/issues/1479",
            "updated_at": "2017-09-07T15:00:32Z",
            "url": "https://api.github.com/repos/cython/cython/issues/comments/327826728",
            "user": {
                "avatar_url": "https://avatars0.githubusercontent.com/u/491659?v=4",
                "events_url": "https://api.github.com/users/scoder/events{/privacy}",
                "followers_url": "https://api.github.com/users/scoder/followers",
                "following_url": "https://api.github.com/users/scoder/following{/other_user}",
                "gists_url": "https://api.github.com/users/scoder/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/scoder",
                "id": 491659,
                "login": "scoder",
                "organizations_url": "https://api.github.com/users/scoder/orgs",
                "received_events_url": "https://api.github.com/users/scoder/received_events",
                "repos_url": "https://api.github.com/users/scoder/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/scoder/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/scoder/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/scoder"
            }
        }
    ],
    "comments": 7,
    "comments_url": "https://api.github.com/repos/cython/cython/issues/1479/comments",
    "created_at": "2016-10-07T19:29:47Z",
    "event_data": [
        {
            "actor": {
                "avatar_url": "https://avatars0.githubusercontent.com/u/491659?v=4",
                "events_url": "https://api.github.com/users/scoder/events{/privacy}",
                "followers_url": "https://api.github.com/users/scoder/followers",
                "following_url": "https://api.github.com/users/scoder/following{/other_user}",
                "gists_url": "https://api.github.com/users/scoder/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/scoder",
                "id": 491659,
                "login": "scoder",
                "organizations_url": "https://api.github.com/users/scoder/orgs",
                "received_events_url": "https://api.github.com/users/scoder/received_events",
                "repos_url": "https://api.github.com/users/scoder/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/scoder/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/scoder/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/scoder"
            },
            "commit_id": null,
            "commit_url": null,
            "created_at": "2017-02-07T06:35:55Z",
            "event": "labeled",
            "id": 951556291,
            "label": {
                "color": "444444",
                "name": "R: duplicate"
            },
            "url": "https://api.github.com/repos/cython/cython/issues/events/951556291"
        },
        {
            "actor": {
                "avatar_url": "https://avatars0.githubusercontent.com/u/491659?v=4",
                "events_url": "https://api.github.com/users/scoder/events{/privacy}",
                "followers_url": "https://api.github.com/users/scoder/followers",
                "following_url": "https://api.github.com/users/scoder/following{/other_user}",
                "gists_url": "https://api.github.com/users/scoder/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/scoder",
                "id": 491659,
                "login": "scoder",
                "organizations_url": "https://api.github.com/users/scoder/orgs",
                "received_events_url": "https://api.github.com/users/scoder/received_events",
                "repos_url": "https://api.github.com/users/scoder/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/scoder/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/scoder/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/scoder"
            },
            "commit_id": null,
            "commit_url": null,
            "created_at": "2017-02-07T06:36:00Z",
            "event": "closed",
            "id": 951556375,
            "url": "https://api.github.com/repos/cython/cython/issues/events/951556375"
        }
    ],
    "events_url": "https://api.github.com/repos/cython/cython/issues/1479/events",
    "html_url": "https://github.com/cython/cython/issues/1479",
    "id": 181749774,
    "labels": [
        {
            "color": "444444",
            "default": false,
            "id": 425556894,
            "name": "R: duplicate",
            "url": "https://api.github.com/repos/cython/cython/labels/R:%20duplicate"
        }
    ],
    "labels_url": "https://api.github.com/repos/cython/cython/issues/1479/labels{/name}",
    "locked": false,
    "milestone": null,
    "number": 1479,
    "repository_url": "https://api.github.com/repos/cython/cython",
    "state": "closed",
    "title": "Cython compilation fails for valid nested generator comprehension.",
    "updated_at": "2017-09-07T15:00:32Z",
    "url": "https://api.github.com/repos/cython/cython/issues/1479",
    "user": {
        "avatar_url": "https://avatars2.githubusercontent.com/u/406380?v=4",
        "events_url": "https://api.github.com/users/gwk/events{/privacy}",
        "followers_url": "https://api.github.com/users/gwk/followers",
        "following_url": "https://api.github.com/users/gwk/following{/other_user}",
        "gists_url": "https://api.github.com/users/gwk/gists{/gist_id}",
        "gravatar_id": "",
        "html_url": "https://github.com/gwk",
        "id": 406380,
        "login": "gwk",
        "organizations_url": "https://api.github.com/users/gwk/orgs",
        "received_events_url": "https://api.github.com/users/gwk/received_events",
        "repos_url": "https://api.github.com/users/gwk/repos",
        "site_admin": false,
        "starred_url": "https://api.github.com/users/gwk/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/gwk/subscriptions",
        "type": "User",
        "url": "https://api.github.com/users/gwk"
    }
}